client/ayon_harmony/api/pipeline.py

```diff
@@ -253,13 +253,14 @@ def ls():
     """
     scene_data = harmony.get_scene_data() or dict()
     all_top_names = harmony.get_all_top_names()
+    palettes_paths = harmony.get_palettes_paths()
     cleaned_scene_data = True
     for entity_name, entity_data in scene_data.copy().items():
         if not is_container_data(entity_data):
             continue
 
         # Filter orphaned containers
-        if entity_name not in all_top_names:
+        if entity_name not in all_top_names | palettes_paths:
```

Comment:
Can we generate the intersection of these two objects to outside the for loop into a new variable? You're now creating a new set on each iteration.